app/controllers/api/v0/api_controller.rb

```diff
@@ -29,6 +29,11 @@ def auth_results
     render json: { status: "ok" }
   end
 
+  def user_qualification_data
+    user = User.find(params.require(:user_id))
+    render json: user.person.personal_records.map(&:to_wcif)
```

Comment:
Attention! This code will currently crash for users that have never competed before.

Those users have never been assigned a WCA ID, so `user.person` will already return `nil`. You just happily proceed to call `.personal_records` on a potentially nil value, which will lead to crashes.